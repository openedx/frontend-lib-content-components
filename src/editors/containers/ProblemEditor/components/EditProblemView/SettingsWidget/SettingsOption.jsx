import React from 'react';
import { Collapsible, Icon, Card } from '@edx/paragon';
import { KeyboardArrowUp, KeyboardArrowDown } from '@edx/paragon/icons';
import PropTypes from 'prop-types';
import { showFullCard } from './hooks';

export const SettingsOption = ({
  title,
  summary,
  none,
  children,
  className,
}) => {
  const { isCardCollapsed, toggleCardCollapse } = showFullCard();

  return (
    <Card className={`${className} settingsOption border border-light-700 shadow-none`}>
      <Card.Section className="settingsCardTitleSection">
        <Collapsible.Advanced
          open={isCardCollapsed}
          onToggle={toggleCardCollapse}
        >
          <Collapsible.Trigger className="collapsible-trigger d-flex">
            <span className="flex-grow-1 text-primary-500 x-small">{title}</span>
            <Collapsible.Visible whenClosed>
              <Icon src={KeyboardArrowDown} />
            </Collapsible.Visible>
            <Collapsible.Visible whenOpen>
              <Icon src={KeyboardArrowUp} />
            </Collapsible.Visible>
          </Collapsible.Trigger>
        </Collapsible.Advanced>
      </Card.Section>
      <Card.Section className="px-4 pb-4 pt-3">
        <Collapsible.Advanced
          open={!isCardCollapsed}
        >
          <Collapsible.Body className="collapsible-body">
            <span className={`small ${none ? 'text-gray-500' : 'text-primary-500'}`}>{summary}</span>
          </Collapsible.Body>
        </Collapsible.Advanced>
        <Collapsible.Advanced
          open={isCardCollapsed}
        >
          <Collapsible.Body className="collapsible-body">
            {children}
          </Collapsible.Body>
        </Collapsible.Advanced>
      </Card.Section>
    </Card>
  );
};

SettingsOption.propTypes = {
  title: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired,
  none: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};
SettingsOption.defaultProps = {
  none: false,
  className: '',
};

export default SettingsOption;
