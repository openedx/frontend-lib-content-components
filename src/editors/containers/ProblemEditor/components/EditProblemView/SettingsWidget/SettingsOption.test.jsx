import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import SettingsOption from './SettingsOption';

describe('SettingsOption', () => {
  describe('default with children', () => {
    const children = (<h1>My test content</h1>);
    test('snapshot: renders correct', () => {
      expect(shallow(<SettingsOption title="Settings Option Title" summary="Settings Option Summary">{children}</SettingsOption>).snapshot).toMatchSnapshot();
    });
  });
  describe('with additional sections', () => {
    const children = (<h1>First Section</h1>);
    const sections = [<h1>Second Section</h1>, <h1>Third Section</h1>];
    test('snapshot: renders correct', () => {
      expect(shallow(
        <SettingsOption title="Settings Option Title" summary="Settings Option Summary" extraSections={sections}>
          {children}
        </SettingsOption>,
      ).snapshot).toMatchSnapshot();
    });
  });
});
