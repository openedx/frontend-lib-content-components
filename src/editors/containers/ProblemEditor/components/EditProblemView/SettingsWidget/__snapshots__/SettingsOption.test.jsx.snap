// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsOption default with children snapshot: renders correct 1`] = `
<Card
  className=" settingsOption border border-light-700 shadow-none"
>
  <Card.Section
    className="settingsCardTitleSection"
    key="settingsOption-Settings Option Title-header"
  >
    <Advanced
      onToggle={[Function]}
      open={false}
    >
      <Trigger
        className="collapsible-trigger d-flex"
      >
        <span
          className="flex-grow-1 text-primary-500 x-small"
        >
          Settings Option Title
        </span>
        <Visible
          whenClosed={true}
        >
          <Icon />
        </Visible>
        <Visible
          whenOpen={true}
        >
          <Icon />
        </Visible>
      </Trigger>
    </Advanced>
  </Card.Section>
  <CardSection
    isCardCollapsibleOpen={false}
    key="settingsOption-Settings Option Title-children"
    none={false}
    summary="Settings Option Summary"
  >
    <h1>
      My test content
    </h1>
  </CardSection>
</Card>
`;

exports[`SettingsOption with additional sections snapshot: renders correct 1`] = `
<Card
  className=" settingsOption border border-light-700 shadow-none"
>
  <Card.Section
    className="settingsCardTitleSection"
    key="settingsOption-Settings Option Title-header"
  >
    <Advanced
      onToggle={[Function]}
      open={false}
    >
      <Trigger
        className="collapsible-trigger d-flex"
      >
        <span
          className="flex-grow-1 text-primary-500 x-small"
        >
          Settings Option Title
        </span>
        <Visible
          whenClosed={true}
        >
          <Icon />
        </Visible>
        <Visible
          whenOpen={true}
        >
          <Icon />
        </Visible>
      </Trigger>
    </Advanced>
  </Card.Section>
  <CardSection
    isCardCollapsibleOpen={false}
    key="settingsOption-Settings Option Title-children"
    none={false}
    summary="Settings Option Summary"
  >
    <h1>
      First Section
    </h1>
  </CardSection>
  <Fragment>
    <CardSection
      isCardCollapsibleOpen={false}
      key="settingsOption-Settings Option Title-0"
      none={false}
      summary={null}
    />
  </Fragment>
  <Fragment>
    <CardSection
      isCardCollapsibleOpen={false}
      key="settingsOption-Settings Option Title-1"
      none={false}
      summary={null}
    />
  </Fragment>
</Card>
`;
